test/jdk/tools/jpackage/helpers/jdk/jpackage/test/TKit.java

```diff
@@ -227,28 +226,20 @@ static void runAdhocTest(ThrowingRunnable action) {
         runTests(List.of(test), Set.of(RunTestMode.FAIL_FAST));
     }
 
-    static Runnable ignoreExceptions(ThrowingRunnable action) {
+    static Runnable ignoreExceptions(ThrowingRunnable<? extends Exception> action) {
         return () -> {
             try {
                 try {
                     action.run();
-                } catch (Throwable ex) {
-                    unbox(ex);
+                } catch (Exception ex) {
+                    throw ExceptionBox.unbox(ex);
                 }
-            } catch (Throwable throwable) {
-                printStackTrace(throwable);
+            } catch (Exception|AssertionError t) {
```

Comment:
Should we use space around `|` or not? I see some old cases when we do `catch (ExceptionBox | InvocationTargetException ex)` and in new cases we do `catch (Exception|AssertionError t)`.